src/main/peripherals/ActionNotDefinedException.java

```diff
@@ -0,0 +1,10 @@
+package peripherals;
+
+public class ActionNotDefinedException extends Exception{
```

Comment:
Good that you use custom Exceptions for custom cases, but it should have a custom message as well then.